pkg/config/tlscfg/cert_watcher_test.go

```diff
@@ -82,36 +82,40 @@ func TestReload(t *testing.T) {
 	require.NoError(t, err)
 	assert.Equal(t, &cert, watcher.certificate())
 
-	// update the content with client certs
+	// Write the client's public key.
 	certData, err = ioutil.ReadFile(clientCert)
 	require.NoError(t, err)
 	err = syncWrite(certFile.Name(), certData, 0644)
 	require.NoError(t, err)
+
+	waitUntil(func() bool {
+		// Logged when the cert is reloaded with mismatching client public key and existing server private key.
+		return logObserver.FilterMessage("Failed to load certificate").
+			FilterField(zap.String("certificate", certFile.Name())).Len() > 0
+	}, 100, time.Millisecond*200)
```

Comment:
nit: instead of 100 * 200ms, I would do 2000 * 10ms - minimize the guaranteed delay, keep total delay the same (absolute # is not particularly important, 20s is probably overkill, but 10s would be fine, enough to outlast any scheduling hiccup).